assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -846,8 +849,56 @@ export function HistoryTypeWidget(props: WidgetParams) {
         <small className="text-red-500">{historyNameError}</small>
       </div>
     </>
-  )
-    ;
+  );
+}
+
+export function HistoryModeWidget(props: WidgetParams) {
+  const options = getSelectOptions(props.schema);
+  const userMaxTokenLimit = concatenate(props.nodeParams["user_max_token_limit"]);
+  const initialHistoryMode = concatenate(props.nodeParams["history_mode"]);
+  const [historyMode, setHistoryMode] = useState(initialHistoryMode || "Summarize");
+   const historyModeHelpTexts: Record<string, string> = {
+    "Summarize": "If the token count exceeds the limit, older messages will be summarized while keeping the last few messages intact.",
+    "Truncate Tokens": "If the token count exceeds the limit, older messages will be removed until the token count is below the limit.",
+    "Max History Length": "The chat history will always be truncated to the last 10 messages. Older messages will not be sent."
```

Comment:
"last 10 messages": this number should be configurable. The ticket mentions a "Max History Length" parameter when using this mode which should be an integer.